functors_task_1/student/include/impl.h

```diff
@@ -40,9 +54,18 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+private:
+    static Coctails m_coctails;
+public:
+    static Cocktail mix(AlcoholDrink alcohol, NonAlcoholDrink nonAlcohol)
     {
+        return m_coctails.find({alcohol, nonAlcohol}) != m_coctails.end() ? m_coctails[{alcohol, nonAlcohol}]
+                                                                          : Cocktail::Oops;
     }
 };
 
-std::function</**???*/> mixer {&MixingPolicy::mix};
+Coctails MixingPolicy::m_coctails= {{{AlcoholDrink::Gin, NonAlcoholDrink::LimeJuice}, Cocktail::Gimlet},
```

Comment:
1. Why do you need to recreate this map each function call?
2. Why do you need ordered container?
3. Please implement this task without containers usage